tests/test_ledger.ml

```diff
@@ -0,0 +1,186 @@
+open Setup
+open Helpers
+open Crypto
+open Core
+open Ledger
+
+let () =
```

Comment:
It is a very big function. 
Could it be possible to split them into smaller functions and then each test dedicated for the kind of test that you want to test, in the end, call those test functions?